connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/MySQLIntegrationSuite.scala

```diff
@@ -241,6 +241,37 @@ class MySQLIntegrationSuite extends DockerJDBCIntegrationV2Suite with V2JDBCTest
     assert(rows10(0).getString(0) === "amy")
     assert(rows10(1).getString(0) === "alex")
   }
+
+  test("SPARK-50793: MySQL JDBC Connector failed to cast some types") {
+    val tableName = catalogName + ".test_cast_function"
+    withTable(tableName) {
+      // CREATE table to use types defined in Spark SQL
+      sql(s"""CREATE TABLE $tableName (
+        string_col STRING,
+        short_col SHORT,
+        integer_col INTEGER,
+        long_col LONG,
+        binary_col BINARY,
+        double_col DOUBLE
+      )""")
+      sql(s"INSERT INTO $tableName VALUES('0', 0, 0, 0, x'30', 0.0)")
+
+      def testCast(castType: String, sourceCol: String, targetCol: String): Unit = {
+        val sql = s"SELECT * FROM $tableName WHERE CAST($sourceCol AS $castType) = $targetCol"
+        assert(spark.sql(sql).collect().length === 1)
```

Comment:
let's test the actual value

Reply:
Sure, let me add that.

Reply:
The test is validating actual value now, PTAL.